x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/support/Hasher.java

```diff
@@ -506,17 +512,21 @@ private static boolean verifyPbkdf2Hash(SecureString data, char[] hash) {
             int cost = Integer.parseInt(new String(Arrays.copyOfRange(hash, PBKDF2_PREFIX.length(), hash.length - (2 * tokenLength + 2))));
             SecretKeyFactory secretKeyFactory = SecretKeyFactory.getInstance("PBKDF2withHMACSHA512");
             PBEKeySpec keySpec = new PBEKeySpec(data.getChars(), Base64.getDecoder().decode(CharArrays.toUtf8Bytes(saltChars)),
-                cost, PBKDF2_KEY_LENGTH);
+                cost, 256);
             char[] computedPwdHash = CharArrays.utf8BytesToChars(Base64.getEncoder()
                 .encode(secretKeyFactory.generateSecret(keySpec).getEncoded()));
-            boolean result = CharArrays.constantTimeEquals(computedPwdHash, hashChars);
+            final boolean result = CharArrays.constantTimeEquals(computedPwdHash, hashChars);
             Arrays.fill(computedPwdHash, '\u0000');
```

Comment:
not a big deal, but maybe we should pull the `computedPwdHash` array out of the try and set it to null initially. Then we can fill it in the finally with the others?

Reply:
I was just trying to zerofill as soon after the use as possible. Do you see any advantages other than consistency?

Reply:
An exception is thrown during the conversion of utf8 bytes to chars would cause this array to hang around in memory. Like I said, if you prefer how you have it, I am fine with it :)

Reply:
Gotcha. Nope, I was just trying to see what you're getting at thanks. I'll change it